method/message/data/createLao.json

```diff
@@ -0,0 +1,110 @@
+{
+    "$comment": "By sending the lao/create message to the organizer’s server’s default channel (“/root”), the main channel of the LAO will be created with the identifier id. At that point, any of the clients, including the organizer, can subscribe to the channel id and send messages over it.",
+    "$comment": "The server is expected to verify the data, such that the last modified timestamp is equal to the creation timestamp, that the timestamp is reasonably recent with respect to the server’s clock, that the attestation is valid, etc.",
+    "$comment": "Upon successful creation, the server/organizer is expected to broadcast the LAO state to all witnesses and clients (see “LAO state broadcast”).",
+    "$schema": "http://json-schema.org/draft/2019-09/schema#",
+    "$id": "http://pop2020.dedis.ch/createLao.json",
+    "type": "object",
+    "properties" : {
+        "jsonrpc": {
+            "const": "2.0"
+        },
+        "method": {
+            "description": "Operation to be performed",
+            "type": "string",
+            "enum": ["publish"]
+        },
+        "params": {
+            "description": "Content of the request",
+            "type": "object",
+            "properties": {
+                "channel": {
+                    "description": "Channel id",
+                    "type": "string"
+                },
+                "message": {
+                    "description": "Content of the message",
+                    "type": "object",
+                    "properties": {
+                        "data": {
+                            "description": "base64 representation of the data",
+                            "type": "object",
+                            "$comment": "must be in base64",
+                            "properties": {
+                                "object": {
+                                    "description": "object on which to perform the action",
+                                    "type": "string",
+                                    "enum": ["lao"]
+                                }, 
+                                "action": {
+                                    "description": "action to perform on the object",
+                                    "type": "string",
+                                    "enum": ["create"]
+                                }, 

```

Comment:
If it's base64 encoded, JSON Schema can't validate it. It'll just have to be a type string.

Reply:
@Huntsekkern defined the folder structure and a few examples. I will create the files themselves. For example, here's the validator for a "message content"

```json
{
	"title": "Match message content of a LAO creation message",
	"description": "Match message content for Witness a message",
	"type": "object",
	"properties": {
		"data": {
			"type": "string",
			"contentEncoding": "base64"
		},
		"sender": {
			"type": "string", 
			"pattern": "^0x[a-fA-F0-9]+$",
			"$comment": "[HexString] public key of the sender"
		},
		"signature": {
			"type": "string", 
			"pattern": "^0x[a-fA-F0-9]+$",
			"$comment": "[HexString] organizer's signature on data : Sign(Hash(data))"
		},
		"message_id": {
			"type": "string", 
			"pattern": "^0x[a-fA-F0-9]+$",
			"$comment": "[HexString] message id : Hash(data||signature)"
		},
		"witness_signatures": {
			"type": "array", 
			"maxItems": 0,
			"$comment": "list of public keys of witnesses : always empty in this case"
		}
	}, 
	"additionalProperties": false,
	"required": ["data", "sender", "signature", "message_id", "witness_signatures"]
}
```

Reply:
As Bryan said, Baes64 is the standard method of store the crypto related things. Could you update it?

Reply:
Yes sure! I already did it. But I was waiting for the pull request to be accepted. Should I also push changes on this branch (work-proto-raoul) @haoqianzhang ?

Reply:
You can merge it. Or push the changes to this branch. Both are fine I think.